### ISSUE TITLE
Allow to add new Enumeration

### ISSUE THREAD

**Issue by AlaJaber-Tahaluf** [microsoft/Power-Fx#683]
as you allowed me to add a defined new Function to power-fx, Like this:
`
            var config = new PowerFxConfig();
            config.AddFunction(new HelpFunction());
            var engine = new RecalcEngine(config);
`

I want to allowed me to defined new Enumeration for example:
`
{
    EnumConstants.GenderEnumString,
    "%s[Male:\"male\", Female:\"female\", Other:\"other\"]"
}
`

**Comment by MikeStall** [comment]
Does this example solve that: https://github.com/microsoft/power-fx-host-samples/blob/107cac581b66366aa3eedb30f4075f915e2a0dcd/Samples/ConsoleREPL/ConsoleREPL.cs#L36

**Comment by AlaJaber-Tahaluf** [comment]
I think that will help me to create new Enum, but it's limited because it takes Only KeyPair<string, string> and I need to be for example KeyPair<string, int>

**Comment by lesaltzm** [comment]
Yes, that's intended. The current Enum implementation is not something we'd like to expand, it's not as type safe as Option Sets, and we'd prefer that any uses without back-compat considerations build on the safer option. 

Do you have a use case where the stringified ints as the "logical" names wouldn't be sufficient? Given that it's typesafe, if you have a custom function that expects an option set value of your option set, you should be able to safely call int.TryParse on the value.
